Services/Sources/GitLabSource.cs

```diff
@@ -15,24 +15,103 @@
 * If not, see https://www.gnu.org/licenses/lgpl-3.0.txt
 */
 
+using Models;
+using Newtonsoft.Json;
+using RestSharp;
+using Services.Sources.Resources;
 using System;
+using System.Text;
+using System.Text.RegularExpressions;
 
 namespace Services.Sources
 {
-
+    /// <summary>
+    /// GitlabSource
+    /// </summary>
+    /// <seealso cref="ISource" />
     public class GitLabSource : ISource
     {
+        /// <summary>
+        /// The gitlab API URL
+        /// </summary>
+        private readonly string gitlabApiUrl = "https://gitlab.com/api/v4/projects/";
 
+        /// <summary>
+        /// Gets the source.
+        /// </summary>
+        /// <param name="url">The URL.</param>
+        /// <exception cref="NotImplementedException"></exception>
         public void GetSource(string url)
         {
             throw new NotImplementedException();
         }
 
+        /// <summary>
+        /// Gets the project information.
+        /// </summary>
+        /// <param name="url">The URL.</param>
+        /// <returns></returns>
+        public Project GetProjectInformation(string url)
+        {
+            url = Regex.Replace(url, @"^((https?):\/\/)?(www.)?gitlab\.com\/", "");
+            string serializedUrl = this.gitlabApiUrl + url.Replace("/", "%2F");
+
+            Project project = new Project();
+
+            GitLabResourceResult resourceResult = FetchRepo(serializedUrl);
+            project.Name = resourceResult.name;
+            project.ShortDescription = resourceResult.description;
+            project.Uri = url;
+            if(!string.IsNullOrEmpty(resourceResult.readme_url))
+            {
+                project.Description = FetchReadme(resourceResult.readme_url);
+            }
+
+            return project;
+        }
+
+        public bool ProjectUrlMatches(string url)
+        {
+            Regex rx = new Regex(@"^((https?):\/\/)?(www.)?gitlab\.com\/.+\/.+$", RegexOptions.Compiled | RegexOptions.IgnoreCase);
+            MatchCollection matches = rx.Matches(url);
+            return rx.IsMatch(url);
+        }
+
+        /// <summary>
+        /// Searches the specified search term.
+        /// </summary>
+        /// <param name="searchTerm">The search term.</param>
+        /// <exception cref="NotImplementedException"></exception>
         public void Search(string searchTerm)
```

Comment:
Search is not yet implemented but a feature that has yet to come, so removing them is not needed imo.

Reply:
I agree that it may seem unneeded to keep the methods in at this point. Looking at the future, when we implemented a real Github/Gitlab wizard (with oauth flow), like we discussed. This will become more useful right? 
If not, we could remove it, but I think it will become useful at that point.

Reply:
Seeing as no one replied in time, we keep it in for now. We need to move forward. Thanks for the input @DigitalExcellence/backend 😓